internal/provider/organization_data_source.go

```diff
@@ -90,12 +91,11 @@ func (d *OrganizationDataSource) Read(ctx context.Context, req datasource.ReadRe
 
 	// we fetch the organization id from the service account
 	orgMrn := ""
-	if data.OrgMrn.ValueString() != "" {
+	if data.OrgMrn.ValueString() != "" && data.OrgID.ValueString() == "" {
 		orgMrn = data.OrgMrn.ValueString()
-	} else if data.OrgID.ValueString() != "" {
+	} else if data.OrgID.ValueString() != "" && data.OrgMrn.ValueString() == "" {
```

Comment:
org mrn must be empty here since it is an else if condition, can you help me understand why we need that check?